File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int total = 0;
        int prevCount = 0;
        for (String row : bank) {
            int currCount = 0;
            for (int i = 0; i < row.length(); i++) {
                if (row.charAt(i) == '1') {
                    currCount++;
                }
            }
            if (currCount == 0) continue;
            total += (prevCount * currCount);
            prevCount = currCount;
        }            
        return total;
    }
}

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: list[str]) -> int:
        total=0
        prevcount=0
        for i in range(len(bank)):
            currcount=bank[i].count('1')
            if(currcount==0):
                continue
            total+=(prevcount*currcount)
            prevcount=currcount
            
        return total